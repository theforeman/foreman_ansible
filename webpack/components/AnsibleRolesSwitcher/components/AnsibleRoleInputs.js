import React from 'react';
import PropTypes from 'prop-types';

const AnsibleRoleInputs = props => {
  const { role, idx, resourceName } = props;

  const fieldName = attr =>
    `${resourceName}[${resourceName}_ansible_roles_attributes][${idx}][${attr}]`;

  const attrName = modelName => `${modelName}AnsibleRoleId`;

  const idField = (
    <input
      name={fieldName('id')}
      value={role[attrName(resourceName)]}
      type="hidden"
    />
  );
  return (
    <React.Fragment>
      {role[attrName(resourceName)] ? idField : null}
      <input
        name={fieldName('ansible_role_id')}
        value={role.id}
        type="hidden"
      />
      <input name={fieldName('position')} value={idx + 1} type="hidden" />
      <input
        name={fieldName('_destroy')}
        value={!!role.destroy}
        type="hidden"
      />
    </React.Fragment>
  );
};

AnsibleRoleInputs.propTypes = {
  role: PropTypes.object.isRequired,
  idx: PropTypes.number.isRequired,
  resourceName: PropTypes.string.isRequired,
};

export default AnsibleRoleInputs;
