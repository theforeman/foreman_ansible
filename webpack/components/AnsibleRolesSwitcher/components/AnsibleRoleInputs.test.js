import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import AnsibleRoleInputs from './AnsibleRoleInputs';

const fixtures = {
  'should render role to add': {
    role: { id: 2, name: 'test.role', hostAnsibleRoleId: 5, position: 2 },
    resourceName: 'host',
    idx: 14,
  },
  'should render role to remove': {
    role: { id: 2, name: 'test.role', hostAnsibleRoleId: 5, destroy: true },
    resourceName: 'host',
    idx: 14,
  },
};

describe('AnsibleRoleInputs', () =>
  testComponentSnapshotsWithFixtures(AnsibleRoleInputs, fixtures));
