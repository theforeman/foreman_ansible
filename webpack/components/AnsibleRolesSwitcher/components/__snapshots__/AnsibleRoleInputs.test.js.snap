// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnsibleRoleInputs should render role to add 1`] = `
<Fragment>
  <input
    name="host[host_ansible_roles_attributes][14][id]"
    type="hidden"
    value={5}
  />
  <input
    name="host[host_ansible_roles_attributes][14][ansible_role_id]"
    type="hidden"
    value={2}
  />
  <input
    name="host[host_ansible_roles_attributes][14][position]"
    type="hidden"
    value={15}
  />
  <input
    name="host[host_ansible_roles_attributes][14][_destroy]"
    type="hidden"
    value={false}
  />
</Fragment>
`;

exports[`AnsibleRoleInputs should render role to remove 1`] = `
<Fragment>
  <input
    name="host[host_ansible_roles_attributes][14][id]"
    type="hidden"
    value={5}
  />
  <input
    name="host[host_ansible_roles_attributes][14][ansible_role_id]"
    type="hidden"
    value={2}
  />
  <input
    name="host[host_ansible_roles_attributes][14][position]"
    type="hidden"
    value={15}
  />
  <input
    name="host[host_ansible_roles_attributes][14][_destroy]"
    type="hidden"
    value={true}
  />
</Fragment>
`;
